stablehlo/tests/interpret/dot_general.mlir

```diff
@@ -29,7 +29,7 @@ func.func @dot_general_op_test_algorithm() {
     algorithm = <
       lhs_precision_type = tf32,
```

Comment:
The dot general interpreter currently uses the result type as the accumulator type 
https://github.com/openxla/stablehlo/blob/8602e098be0aaba038868b943963dd23141eb605/stablehlo/reference/Ops.cpp#L1724

Should we aim to have the interpreter support for some basic cases like lhs/rhs/accumulator type not equal to `tf32` (as I believe types like `tf32` are more like an XLA artifact) and when the component/primitive operations counts are 1.

Reply:
TensorFloat32 is an NVIDIA thing (I also accidentally thought it was tensorflow related), and has upstream MLIR/LLVM support so maybe we can support it as well. I agree that we should make these updates! Maybe in a followup, can make a ticket